github/resource_github_repository_file_test.go

```diff
@@ -297,7 +298,46 @@ func TestAccGithubRepositoryFile_committer(t *testing.T) {
 	})
 }
 
+func TestAccGithubRepositoryFile_overwrite(t *testing.T) {
+	var content github.RepositoryContent
+	var commit github.RepositoryCommit
+
+	rn := "github_repository_file.foo"
+	path := ".gitignore"
+	fileContent := "**/.terraform/*"
+	randString := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	repo := fmt.Sprintf("tf-acc-test-%s", randString)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGithubRepositoryFileDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccGithubRepositoryFileOverwriteDisabledConfig(randString),
+				ExpectError: regexp.MustCompile(`Refusing to overwrite existing file`),
+			},
+			{
+				Config: testAccGithubRepositoryFileOverwriteEnabledConfig(randString, path, fileContent),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGithubRepositoryFileExistsWithRepo(rn, path, "master", repo, &content, &commit),
+					testAccCheckGithubRepositoryFileAttributes(
+						&content, &testAccGithubRepositoryFileExpectedAttributes{
+							Content: base64.StdEncoding.EncodeToString([]byte(fileContent)) + "\n",
+						},
+					),
+				),
+			},
+		},
+	})
+}
+
 func testAccCheckGithubRepositoryFileExists(n, path, branch string, content *github.RepositoryContent, commit *github.RepositoryCommit) resource.TestCheckFunc {
+	hardCodedRepoName := "test-repo"
+	return testAccCheckGithubRepositoryFileExistsWithRepo(n, path, branch, hardCodedRepoName, content, commit)
+}
+
+func testAccCheckGithubRepositoryFileExistsWithRepo(n, path, branch, repo string, content *github.RepositoryContent, commit *github.RepositoryCommit) resource.TestCheckFunc {
```

Comment:
I went down the path of removing references to "test-repo" and leveraging a per-test generated repository instead.  Those changes turned out to break tests so I've opted to leave the existing code paths alone and expose `testAccCheckGithubRepositoryFileExistsWithRepo` instead.  This allows for creating a repository from a template and overwriting its files during the test suite.